src/librustc_mir/shim.rs

```diff
@@ -259,6 +272,164 @@ impl<'a, 'tcx> DropElaborator<'a, 'tcx> for DropShimElaborator<'a, 'tcx> {
     }
 }
 
+/// Build a `Clone::clone` shim for `recvr_ty`. Here, `def_id` is `Clone::clone`.
+fn build_clone_shim<'a, 'tcx>(tcx: ty::TyCtxt<'a, 'tcx, 'tcx>,
+                              def_id: DefId,
+                              rcvr_ty: ty::Ty<'tcx>)
```

Comment:
Actually there are too many variables named `ty` appearing in the code, and I ended up confusing two of them (and having a wrong code generation), that's why I renamed it but I agree `rcvr_ty` is not ideal, I'll find something else.